source/library/CabalGild/Unstable/Action/Render.hs

```diff
@@ -32,26 +34,27 @@ toByteString fs cs =
         $ fields i fs <> comments i cs
 
 -- | Renders the given fields to a block at the given indentation level.
-fields :: Int -> [Fields.Field [Comment.Comment a]] -> Block.Block
+fields :: Int -> [Fields.Field (Position.Position, [Comment.Comment p])] -> Block.Block
 fields = foldMap . field
 
 -- | Renders the given field to a block at the given indentation level.
 --
 -- If a field only has one line and no comments, then it can be rendered all on
 -- one line.
-field :: Int -> Fields.Field [Comment.Comment a] -> Block.Block
+field :: Int -> Fields.Field (Position.Position, [Comment.Comment p]) -> Block.Block
 field i f = case f of
   Fields.Field n fls -> case fls of
     [fl]
-      | null $ FieldLine.annotation fl ->
-          comments i (Name.annotation n)
+      | null . snd $ FieldLine.annotation fl,
+        sameRow (Name.annotation n) (FieldLine.annotation fl) ->
```

Comment:
This is the crux of this PR. Most everything else was refactoring to make this possible.